core/src/main/java/smashdudes/graphics/effects/ParticleEmitter.java

```diff
@@ -120,18 +113,23 @@ private static void updateParticle(ParticleEmitterConfig config, Particle p, flo
         p.life -= dt;
     }
 
-    private static void drawParticle(SpriteBatch sb, Particle p)
+    public boolean depleted()
     {
-        float t = 1.0f  - p.life / p.initialLife;
+        return elapsedTime > config.emissionDuration && activeParticles.size == 0;
+    }
 
-        float scale = MathUtils.lerp(p.scaleStart, p.scaleEnd, t);
+    public void reset()
+    {
+        elapsedTime = 0;
+        spawnTimer = 0;
 
-        float r = MathUtils.lerp(p.rStart, p.rEnd, t);
-        float g = MathUtils.lerp(p.gStart, p.gEnd, t);
-        float b = MathUtils.lerp(p.bStart, p.bEnd, t);
-        float a = MathUtils.lerp(p.aStart, p.aEnd, t);
+        synchronized (particlePool)
+        {
+            particlePool.freeAll(activeParticles);
+            particlePool.freeAll(deadParticles);
+        }
```

Comment:
I don't imagine this needs to be synchronized. 
Since all this function does is give particles back to the pool, it is effectively serial since only one thread can be in the synchronized block at a time.

Reply:
All collections are collections of pointers in java lol.

It is not possible for the pool to have things being obtained at the same time, since things are only added/removed to the pools within update, and the update tasks are joined before proceeding.